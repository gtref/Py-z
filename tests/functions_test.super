// Test file for functions

// Simple function
fn say_hello(name) {
  print("Hello, " + name + "!");
}

say_hello("Super user");


// Function with a return value
fn add(a, b) {
  return a + b;
}

let sum = add(10, 20);
print("10 + 20 =");
print(sum);


// Recursive function (Fibonacci)
fn fib(n) {
  if (n < 2) {
    return n;
  }
  return fib(n - 2) + fib(n - 1);
}

let result = fib(10);
print("Fibonacci(10) =");
print(result); // Should be 55

print("Function test complete.");
