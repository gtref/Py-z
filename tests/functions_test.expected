Hello, Super user!
10 + 20 =
30
Fibonacci(10) =
55
Function test complete.
